xbmc/platform/linux/input/LibInputHandler.cpp

```diff
@@ -93,16 +95,34 @@ CLibInputHandler::CLibInputHandler() : CThread("libinput")
   m_pointer.reset(new CLibInputPointer());
   m_touch.reset(new CLibInputTouch());
   m_settings.reset(new CLibInputSettings(this));
+
+  CServiceBroker::GetAnnouncementManager()->AddAnnouncer(this);
 }
 
 CLibInputHandler::~CLibInputHandler()
 {
+  CServiceBroker::GetAnnouncementManager()->RemoveAnnouncer(this);
   StopThread();
 
   libinput_unref(m_li);
   udev_unref(m_udev);
 }
 
+void CLibInputHandler::Announce(ANNOUNCEMENT::AnnouncementFlag flag, const std::string& sender, const std::string& message, const CVariant& data)
+{
+  if ((flag & (ANNOUNCEMENT::System)))
```

Comment:
I noticed the interior two sets of parenthesis are not required. Please remove the outer one, but I appreciate your use of the inner one to enhance the readability of the long constant, so it can stay: up to you.